src/alpaka/bin/main.cc

```diff
@@ -42,13 +48,20 @@ namespace {
         << std::endl;
   }
 
-  enum class Backend { SERIAL, TBB, CUDA };
+  enum class Backend {
+#ifdef ALPAKA_ACC_CPU_B_SEQ_T_SEQ_SUPPORTED
+    SERIAL,
+#endif
+#ifdef ALPAKA_ACC_CPU_B_TBB_T_SEQ_SUPPORTED
+    TBB,
+#endif
+#ifdef ALPAKA_ACC_GPU_CUDA_SUPPORTED
+    CUDA,
+#endif
```

Comment:
I'd think the `enum` contents (and the code below adding the EDModules based on the selected backend(s)) would not necessarily need the `#ifdef`s, but I don't see a problem with them either.

I'm a bit surprised of C++ allowing a trailing comma in the definition :)

Reply:
> I'm a bit surprised of C++ allowing a trailing comma in the definition :)

me too, but I tested it, and it works...

Reply:
> I'd think the `enum` contents (and the code below adding the EDModules based on the selected backend(s)) would not necessarily need the `#ifdef`s, but I don't see a problem with them either.

Yes, maybe I should add back the definition of the `enum`.

I prefer to keep the EDModules part `#ifdef`ed out, since those symbols do not actually exist if the backend is not enabled.

Reply:
> I prefer to keep the EDModules part `#ifdef`ed out, since those symbols do not actually exist if the backend is not enabled.

Even then having `#ifdef`s for the command line arguments should prevent non-existing EDModules getting loaded (and if not for some reason, the program would terminate with an obscure exception from `std::unordered_map`). Anyway, I'm fine with either way.